v10.6.0 update (#110)
NHSDigital/nhsuk-frontend-jinja: CHANGELOG.md

# NHS.UK frontend jinja changelog

## 0.10.0

This version is compatible with v10.6.0 of nhsuk-frontend.

### :new: **New features**

#### Sortable and scrolling tables

The table component now supports sortable columns and scrolling containers.

- Use `sort: "ascending"` or `sort: "descending"` on the currently sorted column.
- Use `sort: true` on sortable columns that are not currently sorted.

> - Use `scroll: true` for wide tables that need horizontal scrolling.

Reviewer: Sorry this is my fault, I forgot to say we add quotes to all the quote-less Nunjucks examples

```suggestion
- Use `"sort": "ascending"` or `"sort": "descending"` on the currently sorted column.
- Use `"sort": true` on sortable columns that are not currently sorted.
- Use `"scroll": true` for wide tables that need horizontal scrolling.
```

Author: Fixed

- Cell-level `align`, `width`, `href` and `visuallyHiddenText`

This was added in [pull request #1998: Add compact option for tables](https://github.com/nhsuk/nhsuk-frontend/pull/1998), [pull request #2003: Add striped option for tables](https://github.com/nhsuk/nhsuk-frontend/pull/2003), and [pull request #2025: Align table component options with summary list](https://github.com/nhsuk/nhsuk-frontend/pull/2025).

#### Expanded summary list options

Summary lists now support `width`, `id`, and `attributes` on rows and on nested `key`, `value`, and `actions` objects.

This was added in [pull request #2047: Review options for headings and ARIA labels](https://github.com/nhsuk/nhsuk-frontend/pull/2047).

#### Checkbox inclusive option (select all)

Checkbox items now support `behaviour: "inclusive"` to provide a select-all style option in supported browsers.

This was added in [pull request #1707: Add checkbox "all" option](https://github.com/nhsuk/nhsuk-frontend/pull/1707).

#### Card and summary list actions as buttons

Card and summary list action items now support button-style options in addition to link-style options:

- `item.id`
- `item.type`
- `item.name`

> - `item.value`

Reviewer: The changelog has been rewritten 😬

In this case it's summarised the entry and removed the example

```suggestion
#### Use buttons for card and summary list actions

You can now configure card and summary list actions as button elements, using new Nunjucks macro options:

- `item.id` for the element `id` attribute
- `item.type` for the button `type` attribute
- `item.name` for the button `name` attribute
- `item.value` for the button `value` attribute

Action items without `href` will be visually styled as links.

```patch
+ <form method="post" novalidate>
    {{ card({
      "heading": "Regional Manager",
      "actions": {
        "items": [
          {
            "text": "Delete",
-           "href": "/delete"
+           "type": "submit",
+           "name": "action",
+           "value": "delete"
          },
          {
            "text": "Withdraw",
-           "href": "/withdraw",
+           "type": "submit",
+           "name": "action",
+           "value": "withdraw"
          }
        ]
```

Author: Fixed

This was added in [pull request #1989: Add support for card and summary list actions as buttons](https://github.com/nhsuk/nhsuk-frontend/pull/1989).

#### Simpler form option values across components

> Form components now support string values where nested objects were previously required for labels, legends, hints and error messages.

Reviewer: This one's been rewritten too and the example has gone, sorry

```suggestion
#### Use simpler label, legend, hint and error message options

We've updated all form components to support alternative string values for labels, legends, hints and error messages.

For example, when no other nested options are necessary:

```patch
  {{ textarea({
-   "label": {
-     "text": "Can you provide more detail?"
-   },
-   "hint": {
-     "text": "Do not include personal information like your name, date of birth or NHS number"
-   },
+   "label": "Can you provide more detail?",
+   "hint": "Do not include personal information like your name, date of birth or NHS number",
    "name": "more-detail"
  }) }}
```
```

Author: Fixed

> This was added in [pull request #2047: Review options for headings and ARIA labels](https://github.com/nhsuk/nhsuk-frontend/pull/2047).

Reviewer: We should keep "Nunjucks" when it's mentioned in the original PR title

```suggestion
This was added in [pull request #2047: Review Nunjucks options for headings and ARIA labels](https://github.com/nhsuk/nhsuk-frontend/pull/2047).
```

Author: Fixed
